src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -1175,35 +1176,24 @@ class AddItemsPage extends Component {
   saveAndTransitionToNextStep(formValues, lineItems) {
     this.props.showSpinner();
 
-    if (formValues.origin.type === 'SUPPLIER' || !formValues.hasManageInventory) {
-      this.saveRequisitionItems(lineItems)
-        .then((resp) => {
-          let values = formValues;
-          if (resp) {
-            values = { ...formValues, lineItems: resp.data.data.lineItems };
-          }
-          this.transitionToNextStep('CHECKING')
-            .then(() => {
-              this.props.nextPage(values);
-            })
-            .catch(() => this.props.hideSpinner());
-        })
-        .catch(() => this.props.hideSpinner());
-    } else {
-      this.saveRequisitionItems(lineItems)
-        .then((resp) => {
-          let values = formValues;
-          if (resp) {
-            values = { ...formValues, lineItems: resp.data.data.lineItems };
-          }
-          this.transitionToNextStep('VERIFYING')
-            .then(() => {
-              this.props.nextPage(values);
-            })
-            .catch(() => this.props.hideSpinner());
-        })
-        .catch(() => this.props.hideSpinner());
-    }
+    // TODO: This is pre-existing code but why are we using 'VERIFYING' here for non-suppliers?
+    //       It's not a valid StockMovementStatusCode value. Should this be 'VALIDATED'??
+    const status = (formValues.origin.type === 'SUPPLIER' || !formValues.hasManageInventory)
+      ? 'CHECKING' : 'VERIFYING';
```

Comment:
?? I don't get how this is working when we set status to 'VERIFYING'

Reply:
@ewaterman I briefly checked that and you are right this shouldn't work. But from what I can see this code is never executed. Probably after adding approval workflow, we now rely on the `submitRequest` method that executes `saveRequisitionItems` and  `transitionToNextStep`, while the `saveAndTransitionToNextStep` is not used now.